### ISSUE TITLE
Set Tensorflow session seed

### ISSUE THREAD

**Issue by ciberger** [tidymodels/embed#14]
Hi! Thanks for developing such a good (and needed) recipe extension!

I was looking into the documentation for a while and could not find a parameter to set the Tensorflow seed to get reproducible results. Each time I rerun the transformation I get a new seed, I'd instead fix this behaviour at least when transforming the training data set.

Keep it up!
Cris

**Comment by DavisVaughan** [comment]
A reference to this article is mentioned in one sentence at the bottom of the docs for step_embed()! https://keras.rstudio.com/articles/faq.html#how-can-i-obtain-reproducible-results-using-keras-during-development

**Comment by topepo** [comment]
Can you provide a [reprex](https://www.tidyverse.org/help/) where you run the same recipe more than once? Try setting the seed before prepping the recipe. The [underlying code](https://github.com/tidymodels/embed/blob/0672f937d64b8af5bd8b107f2d94d3cf25c1169a/R/tf.R#L244) should set the TF seed(s) based on R's random number stream.

**Comment by ciberger** [comment]
Thanks for your quick replies guys! 

Basically, each time I rerun the embeddings I get a new session seed number, it seems there is no way to modify the seeds param on the _tf_coefs2_ function.

On the other hand, using _use_session_with_seed(...)_ before calling the _step_embed_ function does not work either.

**Test 1**
``` r
library(vcd)
data(Arthritis)

require(recipes)
require(embed)
#> Loading required package: embed
recipe.obj <- recipe(formula("Improved ~ Treatment + Sex + Age"), data = Arthritis) %>%
  step_embed(
    all_nominal(), -all_outcomes(),
    predictors = vars(all_numeric()),
    outcome = vars(Improved),
    num_terms = 2,
    hidden_units = 0,
    options = embed_control(
      epochs = 400, validation_split = .2
    )
  ) %>%
  prep(training = Arthritis)
#> Set session seed to 1695 (disabled GPU, CPU parallelism)
```

<sup>Created on 2019-01-20 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Test 2**
``` r
library(vcd)
data(Arthritis)

require(recipes)
require(embed)
#> Loading required package: embed
recipe.obj <- recipe(formula("Improved ~ Treatment + Sex + Age"), data = Arthritis) %>%
  step_embed(
    all_nominal(), -all_outcomes(),
    predictors = vars(all_numeric()),
    outcome = vars(Improved),
    num_terms = 2,
    hidden_units = 0,
    options = embed_control(
      epochs = 400, validation_split = .2
    )
  ) %>%
  prep(training = Arthritis)
#> Set session seed to 5576 (disabled GPU, CPU parallelism)
```

<sup>Created on 2019-01-20 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by skeydan** [comment]
To me it seems that when 

```
tf_coefs2 <- function(x, y, z, opt, num, lab, h, seeds = sample.int(10000, 4), ...) {
```

gets called, you get  different `seeds` vectors in fresh R sessions, as well as when calling the function  several times in a row:

```
# restart R
sample.int(10000, 4)
sample.int(10000, 4)
## restart R
sample.int(10000, 4)
sample.int(10000, 4)
```

```
> # restart R
> sample.int(10000, 4)
[1] 8528 4771 4492 7039
> sample.int(10000, 4)
[1] 2927 7478  803 7227

Restarting R session...

> # restart R
> sample.int(10000, 4)
[1] 4363 6519 7028 6792
> sample.int(10000, 4)
[1] 7621 7222 9856 2809
```

**Comment by topepo** [comment]
If have never set the seed, you should get different results. 

Since the recipe steps pull the tensorflow seeds from R's random numbers, you only need to set R's seed to get the same TF seeds:
``` r
library(embed)
#> Loading required package: recipes
#> Loading required package: dplyr
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
#> 
#> Attaching package: 'recipes'
#> The following object is masked from 'package:stats':
#> 
#>     step
data(okc)

set.seed(34523)
take_1 <- 
  recipe(Class ~ age + location, data = okc) %>%
  step_embed(location, outcome = vars(Class),
             options = embed_control(epochs = 10)) %>% 
  prep(training = okc) %>% 
  tidy(number = 1)
#> Set session seed to 5504 (disabled GPU, CPU parallelism)

set.seed(34523)
take_2 <- 
  recipe(Class ~ age + location, data = okc) %>%
  step_embed(location, outcome = vars(Class),
             options = embed_control(epochs = 10)) %>% 
  prep(training = okc) %>% 
  tidy(number = 1)
#> Set session seed to 5504 (disabled GPU, CPU parallelism)

all.equal(take_1, take_2)
#> [1] TRUE
```

Created on 2019-01-21 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)

**Comment by skeydan** [comment]
That makes sense!
